### ISSUE TITLE
CHANNELD_AWAITING_LOCKIN after 88 confirmations

### ISSUE THREAD

**Issue by vikbtc** [ElementsProject/lightning#823]
My channel to SLEEPYARK is still not opened even after 87 confirmations. Stopping and restarting the node doesn't help.

```shell
root@lightning:~# lightning-cli listpeers 02f6725f9c1c40333b67faea92fd211c183050f28df32cac3f9d69685fe9665432 debug
{ "peers" : 
	[ 
		{ "id" : "02f6725f9c1c40333b67faea92fd211c183050f28df32cac3f9d69685fe9665432", "connected" : true, "netaddr" : 
			[ "104.198.32.198:9735" ], "channels" : 
			[ 
				{ "state" : "CHANNELD_AWAITING_LOCKIN", "owner" : "lightning_channeld", "short_channel_id" : "506417:720:0", "funding_txid" : "37a707e3acd1c5f9c441be26db48898f9b8383b3990fe458997822715384e93e", "msatoshi_to_us" : 400000000, "msatoshi_total" : 400000000, "dust_limit_satoshis" : 546, "max_htlc_value_in_flight_msat" : 18446744073709551615, "channel_reserve_satoshis" : 0, "htlc_minimum_msat" : 0, "to_self_delay" : 144, "max_accepted_htlcs" : 483 } ], "log" : 
			[ "Peer has reconnected, state CHANNELD_AWAITING_LOCKIN", "Waiting for funding confirmations", "Got depth change 0->88 for 37a707e3acd1c5f9c441be26db48898f9b8383b3990fe458997822715384e93e", "Funding tx 37a707e3acd1c5f9c441be26db48898f9b8383b3990fe458997822715384e93e depth 88 of 3" ] } ] }
```
The is the output from getinfo:
```shell
root@lightning:~# lightning-cli getinfo
{ "id" : "0322154776b0814c78b2675e34c232d91335ce7ad25137a079adfdb9db279429f1", "port" : 9735, "address" : 
	[  ], "version" : "v0.5.2-2016-11-21-1676-gbab3b1a", "blockheight" : 506504, "network" : "bitcoin" }
```

**Comment by lukedemi** [comment]
seeing the exact same issue:

```
{ "peers" :
	[
		{ "id" : "02f6725f9c1c40333b67faea92fd211c183050f28df32cac3f9d69685fe9665432", "connected" : true, "netaddr" :
			[ "104.198.32.198:9735" ], "channels" :
			[
				{ "state" : "CHANNELD_AWAITING_LOCKIN", "owner" : "lightning_channeld", "short_channel_id" : "506410:1784:1", "funding_txid" : "e61aa9944d2832112fd6e39feca7f131d6ceb8cf27959919174951408b818aa6", "msatoshi_to_us" : 440000000, "msatoshi_total" : 440000000, "dust_limit_satoshis" : 546, "max_htlc_value_in_flight_msat" : 18446744073709551615, "channel_reserve_satoshis" : 0, "htlc_minimum_msat" : 0, "to_self_delay" : 144, "max_accepted_htlcs" : 483 } ], "log" :
			[ "Peer has reconnected, state CHANNELD_AWAITING_LOCKIN", "Waiting for funding confirmations", "Got depth change 0->103 for e61aa9944d2832112fd6e39feca7f131d6ceb8cf27959919174951408b818aa6", "Funding tx e61aa9944d2832112fd6e39feca7f131d6ceb8cf27959919174951408b818aa6 depth 103 of 3", "Peer transient failure in CHANNELD_AWAITING_LOCKIN: Owning subdaemon lightning_channeld died (33792)", "Peer has reconnected, state CHANNELD_AWAITING_LOCKIN", "Already have funding locked in" ] } ] }```

getinfo:
```
{ "id" : "03187dab1dd4bfa9b27eccf3ec853e4333921314d7f2adc52a79f1ec89120124f8", "port" : 9735, "address" :
	[
		{ "type" : "ipv4", "address" : "159.89.146.232", "port" : 9735 } ], "version" : "v0.5.2-2016-11-21-1706-gaf26c91", "blockheight" : 506513, "network" : "bitcoin" }
```

**Comment by RvMp** [comment]
I think `to_self_delay` is also applicable when opening a channel. Which are blocks. Takes around 24 hours.

See: https://github.com/lightningnetwork/lightning-rfc/blob/4b5379b2acacada27c7058d2195a88aa1e54193e/05-onchain.md

**Comment by lukedemi** [comment]
That seems wrong, no? I was able to open a channel to TokenSoft.io exactly as expected in 6 confirmations:

```
{ "peers" :
	[
		{ "id" : "0235447c7485ff2b945bac5fbc366d54a87389bab8cacf1b64b26ec01e96bd165a", "connected" : true, "netaddr" :
			[ "34.236.113.58:9735" ], "channels" :
			[
				{ "state" : "CHANNELD_NORMAL", "owner" : "lightning_channeld", "short_channel_id" : "506402:1401:1", "funding_txid" : "e7d307c8dfebca764e9566dda235e0f13dd1547f5671467fa5c3547e5d87ff01", "msatoshi_to_us" : 50000000, "msatoshi_total" : 50000000, "dust_limit_satoshis" : 546, "max_htlc_value_in_flight_msat" : 18446744073709551615, "channel_reserve_satoshis" : 0, "htlc_minimum_msat" : 0, "to_self_delay" : 144, "max_accepted_htlcs" : 483 } ], "log" :
			[ "Peer has reconnected, state CHANNELD_NORMAL", "Already have funding locked in", "Got depth change 0->111 for e7d307c8dfebca764e9566dda235e0f13dd1547f5671467fa5c3547e5d87ff01", "Funding tx e7d307c8dfebca764e9566dda235e0f13dd1547f5671467fa5c3547e5d87ff01 depth 111 of 3", "Funding tx confirmed, but peer state CHANNELD_NORMAL lightning_channeld", "got revoke 78: 0 changed", "got commitsig 79: feerate 20623, 0 added, 0 fulfilled, 0 failed, 0 changed", "got revoke 79: 0 changed", "got commitsig 80: feerate 20622, 0 added, 0 fulfilled, 0 failed, 0 changed" ] } ] }
```

**Comment by dpgribkov** [comment]
I can confirm the same issue @lukedemi having. Channel with SLEEPYARK stuck on `CHANNELD_AWAITING_LOCKIN` after 127 confirmations. And channel to TokenSoft.io is opened after 3 confirmations: 
```
{ "peers" :
	[
		{ "id" : "02f6725f9c1c40333b67faea92fd211c183050f28df32cac3f9d69685fe9665432", "connected" : true, "netaddr" :
			[ "104.198.32.198:9735" ], "channels" :
			[
				{ "state" : "CHANNELD_AWAITING_LOCKIN", "owner" : "lightning_channeld", "short_channel_id" : "506399:2176:1", "funding_txid" : "5aebcbb1e85809f23b073a053c97b7dc6bda674000f0dbb6b40bdd9ad3e8e449", "msatoshi_to_us" : 50000000, "msatoshi_total" : 50000000, "dust_limit_satoshis" : 546, "max_htlc_value_in_flight_msat" : 18446744073709551615, "channel_reserve_satoshis" : 0, "htlc_minimum_msat" : 0, "to_self_delay" : 144, "max_accepted_htlcs" : 483 } ] },
		{ "id" : "0235447c7485ff2b945bac5fbc366d54a87389bab8cacf1b64b26ec01e96bd165a", "connected" : true, "netaddr" :
			[ "34.236.113.58:9735" ], "channels" :
			[
				{ "state" : "CHANNELD_NORMAL", "owner" : "lightning_channeld", "short_channel_id" : "506525:617:1", "funding_txid" : "58b57f3e8b6a65406aae3780cb70c168d5dfa9486f1b32a2724b83093ccc127b", "msatoshi_to_us" : 30000000, "msatoshi_total" : 30000000, "dust_limit_satoshis" : 546, "max_htlc_value_in_flight_msat" : 18446744073709551615, "channel_reserve_satoshis" : 0, "htlc_minimum_msat" : 0, "to_self_delay" : 144, "max_accepted_htlcs" : 483 } ] } ] }
```

**Comment by deksprime** [comment]
Same here. Maybe there's wrong with this SLEEPYARRK guy and his computer won't accept our nodes.
```
 { "peers" : 
	[ 
		{ "id" : "02f6725f9c1c40333b67faea92fd211c183050f28df32cac3f9d69685fe9665432", "connected" : true, "netaddr" : 
			[ "104.198.32.198:9735" ], "channels" : 
			[ 
				{ "state" : "CHANNELD_AWAITING_LOCKIN", "owner" : "lightning_channeld", "short_channel_id" : "506514:1237:1", "funding_txid" : "3354defb4c8de623a77b078a1a8f3e1f57a3e812f32fe50e1c32e1db36f22cb7", "msatoshi_to_us" : 200000000, "msatoshi_total" : 200000000, "dust_limit_satoshis" : 546, "max_htlc_value_in_flight_msat" : 18446744073709551615, "channel_reserve_satoshis" : 0, "htlc_minimum_msat" : 0, "to_self_delay" : 144, "max_accepted_htlcs" : 483 } ] } ] }
 ``` 

Is it possible that this node is having some troubles? Total noob here but maybe after that medium.com tutorial people may have congested his network somehow and now we are having troubles.

**Comment by vikbtc** [comment]
Yeah, I was able to open a channel to TokenSoft.io, although I got a different error when trying to pay an invoice: "failed: WIRE_PERMANENT_NODE_FAILURE (reply from remote)"

I believe node SLEEPYARRK/02f6725f9c1c... is the Blockstream store itself; can someone at Blockstream see what's happening on their end?

**Comment by sfmcnally** [comment]
Same issue with SLEEPYARK

**Comment by RvMp** [comment]
https://github.com/ElementsProject/lightning/issues/815

**Comment by cdecker** [comment]
It's an incentive to open indirect channels to the shop :wink:

We had a bit of an issue keeping up with the blockchain and verifying channels from the gossip sync at the same time. Will need to optimize those lookups and track the UTXO set internally.

**Comment by lukedemi** [comment]
@cdecker what does this mean for users who have channels stuck in CHANNELD_AWAITING_LOCKIN? Should we try to close the channel? Wait it out?

**Comment by lukedemi** [comment]
Looks like the channel is open now! Thanks blockstream ;)

**Comment by cdecker** [comment]
The node should have caught up with the blockchain and the channel state moved onward to either be closed or in normal state. We have a confirmation from @lukedemi, can @vikbtc @dpgribkov and @sfmcnally also confirm that the state has switched?

**Comment by dpgribkov** [comment]
Yes, I can confirm that the state changed from my end as well. I have `CHANNELD_NORMAL` status with SLEEPYARRK/02f6725f9c1c... node.

**Comment by cdecker** [comment]
@sfmcnally `ONCHAIND_OUR_UNILATERAL` means that you started a unilateral close, which will resolve after 144 blocks, i.e., 1 day. Can you ping us here and tell us whether the funds are returned correctly?

**Comment by sfmcnally** [comment]
@cdecker will do

**Comment by fenix2222** [comment]
I keep getting "WIRE_PERMANENT_NODE_FAILURE" when trying to pay for blockstream store stickers. State `connected: true` and `CHANNELD_NORMAL`, but store payment keeps failing

**Comment by cdecker** [comment]
@fenix2222 you're probably connecting through another node that lost its connection. This is the same issue as #550 and will be fixed by #638 

Closing since the original issue seems to be resolved.

**Comment by sfmcnally** [comment]
@cdecker the funds were returned
